folding-schemes/src/folding/hypernova/decider_eth.rs

```diff
@@ -307,6 +301,193 @@ pub mod tests {
+
+        // decider proof generation
+        let proof = D::prove(rng, decider_pp, hypernova.clone()).unwrap();
+
+        let verified = D::verify(
+            decider_vp.clone(),
+            hypernova.i.clone(),
+            hypernova.z_0.clone(),
+            hypernova.z_i.clone(),
+            &(),
+            &(),
+            &proof,
+        )
+        .unwrap();
+        assert!(verified);
+
+        // The rest of this test will serialize the data and deserialize it back, and use it to
+        // verify the proof:
+
+        // serialize the verifier_params, proof and public inputs
+        let mut decider_vp_serialized = vec![];
+        decider_vp
+            .serialize_compressed(&mut decider_vp_serialized)
+            .unwrap();
+        let mut proof_serialized = vec![];
+        proof.serialize_compressed(&mut proof_serialized).unwrap();
+        // serialize the public inputs in a single packet
+        let mut public_inputs_serialized = vec![];
+        hypernova
+            .i
+            .serialize_compressed(&mut public_inputs_serialized)
+            .unwrap();
+        hypernova
+            .z_0
+            .serialize_compressed(&mut public_inputs_serialized)
+            .unwrap();
+        hypernova
+            .z_i
+            .serialize_compressed(&mut public_inputs_serialized)
+            .unwrap();
+        hypernova
+            .U_i
+            .serialize_compressed(&mut public_inputs_serialized)
+            .unwrap();
+        hypernova
+            .u_i
+            .serialize_compressed(&mut public_inputs_serialized)
+            .unwrap();
+
+        // deserialize back the verifier_params, proof and public inputs
+        let decider_vp_deserialized =
+            VerifierParam::<
+                Projective,
+                <KZG<'static, Bn254> as CommitmentScheme<Projective>>::VerifierParams,
+                <Groth16<Bn254> as SNARK<Fr>>::VerifyingKey,
+            >::deserialize_compressed(&mut decider_vp_serialized.as_slice())
+            .unwrap();
+
+        let proof_deserialized =
+            Proof::<Projective, KZG<'static, Bn254>, Groth16<Bn254>>::deserialize_compressed(
+                &mut proof_serialized.as_slice(),
+            )
+            .unwrap();
+
+        let mut reader = public_inputs_serialized.as_slice();
+        let i_deserialized = Fr::deserialize_compressed(&mut reader).unwrap();
+        let z_0_deserialized = Vec::<Fr>::deserialize_compressed(&mut reader).unwrap();
+        let z_i_deserialized = Vec::<Fr>::deserialize_compressed(&mut reader).unwrap();
+        let _U_i = LCCCS::<Projective>::deserialize_compressed(&mut reader).unwrap();
+        let _u_i = CCCS::<Projective>::deserialize_compressed(&mut reader).unwrap();
+
+        let verified = D::verify(
+            (
+                decider_vp_deserialized.pp_hash,
+                decider_vp_deserialized.snark_vp,
+                decider_vp_deserialized.cs_vp,
+            ),
```

Comment:
How would you see updating the Decider's type `VerifierParam` to instead of being (at line 90)
```rust
    type VerifierParam = (C1::ScalarField, S::VerifyingKey, CS1::VerifierParams);
```
being
```rust
    type VerifierParam = VerifierParam<C1, CS1::VerifierParams, S::VerifyingKey>;
```
having imported on the top of the file:
`use crate::folding::nova::decider_eth::VerifierParam;`

So that in this way the call to `D::verify` can be simplified to
```suggestion
        let verified = D::verify(
                decider_vp_deserialized,
```
?

Reply:
(to be clear, the suggestion is to get rid of the tuple in the VerifierParam type, and use directly the Nova's DeciderEth VerifierParam struct so you don't need to convert between the tuple and struct depending on the method call)

Reply:
Yup,  done! [67fcb0f](https://github.com/privacy-scaling-explorations/sonobe/pull/159/commits/67fcb0faec6367ce91922a83fe00660b933a4dc9)